### PR TITLE
Add Reflection parameter

### LINKED ISSUE #35: As a head tutor using the app, I am able to categorise my contacts into different groups.
So that I am clear on the different tutorial groups and their tutors.

### LINKED ISSUE #139: Create new parameter for Reflection Class

### PR DESCRIPTION
**Pull request by jayllo-c**
Currently, we have most of the student information such as name, email,
number and matriculation information, which is sufficient for
broadcasting messages to a large number of people.

However, if we want only want to contact specific reflection classes for
reasons such as class rescheduling, and other targeted announcements or
communication, it would be time-consuming to contact each individual
student.

Let's create a reflection class parameter to associate each student with
a recitation class.

In doing so, we can use the `find` and `copy` functions to quickly contact
the students in specific reflection classes.

Unit testing has also been completed for new and edited code.

Resolves #35, resolves #139

### CODE REVIEW

File: data/avengersassemble.json
Comment (Zer0Legion) [inline data/avengersassemble.json:1]: I think this can be gitignored

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 1]
```diff
@@ -10,12 +10,13 @@
...
 
 import seedu.address.commons.exceptions.IllegalValueException;
-import seedu.address.model.matric.Matric;
 import seedu.address.model.person.Address;
 import seedu.address.model.person.Email;
 import seedu.address.model.person.Name;
 import seedu.address.model.person.Person;
 import seedu.address.model.person.Phone;
+import seedu.address.model.student.Matric;
+import seedu.address.model.student.Reflection;
```
Comment (Zer0Legion) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:19]: Good idea to create a new student package!

**Review by Zer0Legion — COMMENTED** [review]
looks good to me!

**Review by delishad21 — COMMENTED** [review]
Looks Good to me! but we should either remove or gitignore the save file

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 2]
```diff
@@ -69,6 +72,14 @@ private static Matric handleOptionalMatric(ArgumentMultimap argMultimap) throws
         }
     }
 
+    private static Reflection handleOptionalReflection(ArgumentMultimap argMultimap) throws ParseException {
```
Comment (Pughal77) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:75]: Nice abstraction to handle an optional parameter!

**Review by Pughal77 — COMMENTED** [review]
Nice this implementation is similar to the implementation of studio parameter.

**Pull request merged**